srml/authority-discovery/src/lib.rs

```diff
@@ -55,79 +46,40 @@ decl_module! {
 }
 
 impl<T: Trait> Module<T> {
-	/// Returns own authority identifier iff it is part of the current authority
-	/// set, otherwise this function returns None. The restriction might be
-	/// softened in the future in case a consumer needs to learn own authority
-	/// identifier.
-	fn authority_id() -> Option<T::AuthorityId> {
-		let authorities = Keys::<T>::get();
-
-		let local_keys = T::AuthorityId::all();
-
-		authorities.into_iter().find_map(|authority| {
-			if local_keys.contains(&authority) {
-				Some(authority)
-			} else {
-				None
-			}
-		})
-	}
-
 	/// Retrieve authority identifiers of the current authority set.
-	pub fn authorities() -> Vec<T::AuthorityId> {
-		Keys::<T>::get()
-	}
-
-	/// Sign the given payload with the private key corresponding to the given authority id.
-	pub fn sign(
-		payload: &Vec<u8>,
-	) -> Option<(
-		<<T as Trait>::AuthorityId as RuntimeAppPublic>::Signature,
-		T::AuthorityId,
-	)> {
-		let authority_id = Module::<T>::authority_id()?;
-		authority_id.sign(payload).map(|s| (s, authority_id))
+	pub fn authorities() -> Vec<AuthorityId> {
```

Comment:
This function and `Self::keys()` do the same thing. Do you need both?

Could you just do `get(fn authorities)` in the storage above?

Reply:
I am fine with either way. My reasoning of introducing of using a function here is to not expose any internal implementation details. Let's say that we would like to keep track of the current *authority discovery* authority set in the future, it would only need a change within this module and not a change outside of it. Does that make sense @shawntabrizi?

Reply:
I see. So there is a world where maybe `authorities` returns something different than the strict storage of `Keys`.

It makes sense.